### PR TITLE
Enhanced logging around IO actions [BW-413]

### PR DESCRIPTION
**Pull request by salonishah11**
This PR extracts logging from [log_enhanced_53_hotfix](https://github.com/broadinstitute/cromwell/compare/53_hotfix...log_enhanced_53_hotfix) branch that we want in production. 

Closes https://broadworkbench.atlassian.net/browse/BW-413

### CODE REVIEW

File: cwl/src/test/scala/cwl/internal/EnhancedRhinoSandboxSpec.scala [hunk 1]
```diff
@@ -66,7 +66,7 @@ object EnhancedRhinoSandboxSpec extends ScaledTimeSpans {
                              threadReadyLatch: CountDownLatch,
                              threadGoLatch: CountDownLatch): Unit = {
     val cls = ecmaScriptUtilClassLoader.loadClass(classOf[LatchedRunner].getName)
-    val obj = cls.newInstance
+    val obj = cls.getDeclaredConstructor().newInstance()
```
Comment (salonishah11) [inline cwl/src/test/scala/cwl/internal/EnhancedRhinoSandboxSpec.scala:69]: I updated this as compiler was complaining that `.newInstance` is deprecated

File: supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/action/ActionUtils.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ object ActionUtils {
   val sshEntryPoint = "ssh-server"
 
   /** Port mappings for the ssh container. */
-  val sshPortMappings = Map("22" -> new Integer(22))
+  val sshPortMappings = Map("22" -> Integer.valueOf(22))
```
Comment (salonishah11) [inline supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/action/ActionUtils.scala:13]: Similarly here the compiler was complaining that `new Integer(n)` is deprecated and `Integer.valueOf(n)` should be used instead

**Review by cjllanwarne — COMMENTED** [review]
It looks like there are a couple of extra commits on https://github.com/broadinstitute/cromwell/pull/6003 which might also help here?

**Comment by aednichols** [comment]
My review was requested but I also see the "don't look at me" label, skipping for now

File: supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/action/ActionUtils.scala [hunk 3]
```diff
@@ -10,7 +10,7 @@ object ActionUtils {
   val sshEntryPoint = "ssh-server"
 
   /** Port mappings for the ssh container. */
-  val sshPortMappings = Map("22" -> new Integer(22))
+  val sshPortMappings = Map("22" -> Int.box(22))
```
Comment (salonishah11) [inline supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/action/ActionUtils.scala:13]: The compiler was complaining that `new Integer(n)` is deprecated. Hence updated to a more 'scala-esque' option as suggested [here](https://github.com/broadinstitute/cromwell/pull/6003#discussion_r515270682).

**Comment by salonishah11** [comment]
This PR is now ready for review @cjllanwarne @aednichols

**Review by aednichols — COMMENTED** [review]
Changelog (lots of people have asked for this)

**Review by aednichols — APPROVED** [review]

File: filesystems/s3/src/main/scala/cromwell/filesystems/s3/batch/S3BatchIoCommand.scala [hunk 4]
```diff
@@ -99,6 +101,7 @@ sealed trait S3BatchHeadCommand[T] extends S3BatchIoCommand[T, HeadObjectRespons
   */
 case class S3BatchSizeCommand(override val file: S3Path) extends IoSizeCommand(file) with S3BatchHeadCommand[Long] {
   override def mapResponse(response: HeadObjectResponse): Long = response.contentLength
+  override def commandDescription: String = s"S3BatchSizeCommand file '$file'"
```
Comment (cjllanwarne) [inline filesystems/s3/src/main/scala/cromwell/filesystems/s3/batch/S3BatchIoCommand.scala:104]: TOL one could probably move this `def` up into the trait and avoid having to implement it in every sub-class. OTOH I wouldn't suggest going back and retrospectively doing this:
 
```
override def commandDescription: String = s"${getClass.getSimpleName} file '$file'"
```

**Review by cjllanwarne — APPROVED** [review]

File: filesystems/s3/src/main/scala/cromwell/filesystems/s3/batch/S3BatchIoCommand.scala (hunk 4 above)
Comment (salonishah11) [reply filesystems/s3/src/main/scala/cromwell/filesystems/s3/batch/S3BatchIoCommand.scala:104]: That is a valid point. Do you have any strong objections to the current approach being merged as it is?

Comment (cjllanwarne) [reply filesystems/s3/src/main/scala/cromwell/filesystems/s3/batch/S3BatchIoCommand.scala:104]: Yeah - merging as-is is fine with me

**Pull request merged**